### DISCUSSION TITLE
Standard Semantic Search with PGVector + HNSW timing out with 66,000 rows of vector(1536)

### DISCUSSION THREAD

**Discussion by lukenas** [supabase/supabase#18620] [Questions]
Hey, 

I read few of the supabase blog posts about HNSW and PGvector for semantic search. The speed and possibilities got me really excited. I had this query implemented in my supabase project when i only had about 300 vectors and it worked. But now that my database has grown substantially, I've needed to add HNSW indexes. I have now added the index, but the query still times out. Am I missing something? 

CREATE OR REPLACE FUNCTION get_live_opps_semantic(query_embedding vector(1536), similarity_threshold float)
RETURNS TABLE (
    notice_id text
) AS $$
DECLARE
BEGIN
    RETURN QUERY
    SELECT
      dp.notice_id
    FROM 
      live_opps_doc_parts dp
    WHERE 
      1 - (dp.embeddings <=> query_embedding) > similarity_threshold;
END;
$$ LANGUAGE plpgsql;

**Comment by gregnr** [comment]
Hey @lukenas, thanks for posting. As a starting point, can you confirm the operator and column used in your HNSW index?

```sql
select *
from pg_indexes
where tablename = 'live_opps_doc_parts'
```
or from studio:

https://supabase.com/dashboard/project/_/database/indexes

and click 'view definition' next to your `live_opps_doc_parts` HNSW index.

Based on your query, the operator should be `vector_cosine_ops` (when using cosine distance `<=>`) and column should be `embeddings`.
